core/commands/resolve.go

```diff
@@ -78,6 +84,16 @@ Resolve the value of another name recursively:
 			depth = namesys.DefaultDepthLimit
 		}
 
+		if strings.HasPrefix(name, "/ipfs/") || !strings.HasPrefix(name, "/") {
+			resolved, err := resolveIpfsPath(req.Context(), n.Resolver, name)
+			if err != nil {
+				res.SetError(err, cmds.ErrNormal)
+				return
+			}
+			res.SetOutput(&ResolvedPath{resolved})
+			return
+		}
```

Comment:
this needs to go into a different function that calls both: `n.Namesys.ResolveN` and this, because currently, if i point an IPNS name to, say `/ipfs/HASH/foo/bar`, i'll get

```
> ipfs resolve /ipns/NAME
/ipfs/HASH/foo/bar
```
instead of 
```
> ipfs resolve /ipns/NAME
/ipfs/HASH_OF_BAR
```